### PR TITLE
feat: fetch question data from db

### PR DESCRIPTION
**Pull request by PuneetP16**
feature: fetch data from firestore realtime db and render on homepage and category page.

### CODE REVIEW

**Review by saurabhchirde — APPROVED** [review]
Clean and Bold UI, Couldn't find any issue in your code, Looks good to merge.

File: src/pages/Home/Home.jsx [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
+	const trendingQuizCategory = allQuizQuestions.filter(
+		(category) => category.trending
+	);
+
+	const popularQuiz = popularQuizCategory.map((category) => (
+		<CategoryCard key={category._id} category={category} isPopular={true} />
+	));
+	const trendingQuiz = trendingQuizCategory.map((category) => (
+		<CategoryCard key={category._id} category={category} />
+	));
```
Comment (m26unkwn) [inline src/pages/Home/Home.jsx:23]: try to extract the function from here

File: src/utils/firebaseServices.js [hunk 2]
```diff
@@ -180,3 +184,26 @@ export const passwordResetHandler = (email, setAlert) => {
...
+	allQuizQuestions,
+	setAllQuizQuestions
+) => {
+	const dbRef = realTimeDBRef(firebaseRealtimeDB);
+	try {
+		const allQuestions = await get(child(dbRef, "quizDb"));
+		setAllQuizQuestions(allQuestions.val());
+
+		if (allQuizQuestions.length > 1) {
+			localStorage.setItem(ROUTES.quizQuestionData, allQuestions.val());
```
Comment (m26unkwn) [inline src/utils/firebaseServices.js:199]: this is good that you made this kind of stuff using objects

**Review by m26unkwn — APPROVED** [review]
You can go ahead with code

File: src/pages/Home/Home.jsx (hunk 1 above)
Comment (PuneetP16) [reply src/pages/Home/Home.jsx:23]: already extracted from jsx.

File: src/utils/firebaseServices.js (hunk 2 above)
Comment (PuneetP16) [reply src/utils/firebaseServices.js:199]: Thanks

**Pull request merged**